bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsHash.scala

```diff
@@ -56,18 +56,17 @@ object AgsHash {
       c.cc.## +=: c.iq.## +=: c.sb.## +=: buf
     }
 
-    // Base Position
-    Option(ctx.getTargets).foreach { t =>
+    // Asterism
+    Option(ctx.getTargets).foreach { env =>
       val time = Some(new java.lang.Long(when)).asGeminiOpt
-      val base = t.getBase
-
-      def toData(coord: GemOption[java.lang.Double]): Int =
-        coord.asScalaOpt.map(_.doubleValue).##
-
-      val ra  = toData(base.getRaDegrees(time))
-      val dec = toData(base.getDecDegrees(time))
-
-      ra +=: dec +=: buf
+      def hashTarget(t: SPTarget): Unit = {
+        def toData(coord: GemOption[java.lang.Double]): Int =
+          coord.asScalaOpt.map(_.doubleValue).##
+        val ra  = toData(t.getRaDegrees(time))
+        val dec = toData(t.getDecDegrees(time))
+        ra +=: dec +=: buf
+      }
+      env.getAsterism.targets.foreach(hashTarget)
     }
```

Comment:
I'm kind of assuming we want to redo AGS if any component of the asterism moves (as opposed to only looking at movement of the center point). I think in every real-life case these will be equivalent, but I'm trying to anticipate corner cases. I guess it depends on what the GHOST strategy is sensitive to.